airflow/models/dag.py

```diff
@@ -3375,9 +3374,8 @@ def deactivate_deleted_dags(cls, alive_dag_filelocs: list[str], session=NEW_SESS
 
         dag_models = session.query(cls).all()
         for dag_model in dag_models:
-            if dag_model.fileloc is not None:
-                if correct_maybe_zipped(dag_model.fileloc) not in alive_dag_filelocs:
-                    dag_model.is_active = False
+            if dag_model.fileloc is not None and dag_model.fileloc not in alive_dag_filelocs:
+                dag_model.is_active = False
```

Comment:
This change doesn't seem necessary. WDYT?

Reply:
I've added an assertion to an existing test which fails if this change is not also added.

The zipped paths in `alive_dag_filelocs` previously just contained the zip files themselves, e.g. `/opt/airflow/tests/dags/test_zip.zip`. Now they contain the full path, e.g. `/opt/airflow/tests/dags/test_zip.zip/test_zip.py`.

`dag_model.fileloc` has the full path within the zip, so previously `correct_maybe_zipped(dag_model.fileloc)` would match with `/opt/airflow/tests/dags/test_zip.zip` within `alive_dag_filelocs`. Now that `alive_dag_filelocs` contains the full path, we can compare with `dag_model.fileloc` directly.